Created standard issue form for feasibility

jdingeman opened the pull request:
Fixes #2298

### What changes did you make and why did you make them ?

  - Added feasibility-issue-form.yml to have a standard issue form template 
  
### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)
No visual changes to website.

jdingeman committed:
Created new issue form for feasibility

Added new .yml file with issue form template

jdingeman committed:
Update feasibility-issue-template.yml

jdingeman commented:
I was not sure how to test this out within my local repository to view it. If anyone could show me how that'd be awesome!

arpitapandya commented:
Availability: 6 Hours
ETA: EOD 2022-07-29

arpitapandya reviewed (changes requested):
@jdingeman Can you also check mark the Action items in the original issue based on which actions completed or not completed? please!

arpitapandya reviewed (changes requested):
- I would create a unique label above the line options (in the case of checkboxes and dropdowns) and remove key label(3x) with only instruction examples(3x). for e.g. after line 28 you can add a line with label and then remove key label on line 30,31,32. Same thing you can do for the checkboxes for feasibility-items because as per the guide you may want to avoid duplication error so you can also set that as array [reference](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/common-validation-errors-when-creating-issue-forms#bodyi-options-must-be-unique)
- on Line 24 can you fix the typo to attributes?
- on Line 36 can you fix the typo to checklist?
- on line 15 I think I would select the type value textarea(if someone have to write 2 sentences, I think input will allow one line(allows to add first name, last name )
- I went through the guide, syntax you have used are correct.

arpitapandya reviewed (changes requested):
- I would create a unique label above the line options (in the case of checkboxes and dropdowns) and remove key label(3x) with only instruction examples(3x). for e.g. after line 28 you can add a line with label and then remove key label on line 30,31,32. Same thing you can do for the checkboxes for feasibility-items because as per the guide you may want to avoid duplication error so you can also set that as array [reference](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/common-validation-errors-when-creating-issue-forms#bodyi-options-must-be-unique)
- on Line 24 can you fix the typo to attributes?
- on Line 36 can you fix the typo to checklist?
- on line 15 I think I would select the type value textarea(if someone have to write 2 sentences, I think input will allow one line(allows to add first name, last name )
- I went through the guide, syntax you have used are correct.

arpitapandya reviewed (dismissed):
@jdingeman You may have to add explanation to dependency label as its mentioned in the issue (a check box to indicate yes/no dependency, then an input for what it is if yes) for e.g.
```
body:
  - type: dropdown
    id: dependency
    attributes:
      label: Is there a dependency?
      options:
        - "Yes"
        - "No"
    validations:
      required: true
  - type: textarea
    id: dependency-explanation
    attributes:
      label: If Yes, please explain

```
Note: in my example above I am not sure if using yes or no is advisable as per the [guide](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/common-validation-errors-when-creating-issue-forms#forbidden-keys)

arpitapandya commented on .github/ISSUE_TEMPLATE/feasibility-issue-template.yml:
```diff
@@ -0,0 +1,54 @@
+    aattributes:
+      label: "## Action Items"
+      description: "List the research to be done, or the steps to be completed. Note: If the steps can be divided into tasks for more than one person, we recommend dividing it up into separate issues, or assigning it as a pair programming task."
+  - type: checkboxes
```
Sorry, you can also keep the options label for e.g.`    options: 
      - label: Instruction example 1
      - label: Instruction example 2
      - label: Instruction example 3` or `options:
      - Instruction example 1
      - Instruction example 2
      - Instruction example 3` both are fine
      but it will require to have a unique label with a string for the options(which is missing).

arpitapandya commented:
> * I would create a unique label above the line options (in the case of checkboxes and dropdowns) and remove key label(3x) with only instruction examples(3x). for e.g. after line 28 you can add a line with label and then remove key label on line 30,31,32. Same thing you can do for the checkboxes for feasibility-items because as per the guide you may want to avoid duplication error so you can also set that as array [reference](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/common-validation-errors-when-creating-issue-forms#bodyi-options-must-be-unique)
> * on Line 24 can you fix the typo to attributes?
> * on Line 36 can you fix the typo to checklist?
> * on line 15 I think I would select the type value textarea(if someone have to write 2 sentences, I think input will allow one line(allows to add first name, last name )
> * I went through the guide, syntax you have used are correct.

Sorry, you can also keep the options label for e.g. `options: - label: Instruction example 1 - label: Instruction example 2 - label: Instruction example 3` or `options: - Instruction example 1 - Instruction example 2 - Instruction example 3` both are fine
but it will require to have a unique label with a string for the options(which is missing).

jdingeman committed:
update to feasibility-issue-template.yml

jdingeman commented:
Hello @arpitapandya, I've made the changes to the best of my understanding. Please let me know if I have misunderstood anything. Thank you!!

arpitapandya commented:
> Hello @arpitapandya, I've made the changes to the best of my understanding. Please let me know if I have misunderstood anything. Thank you!!

@jdingeman Thanks for the changes. Can you fix the indentation error coming from line 45 so that we can preview it.
![ymlSyntaxError](https://user-images.githubusercontent.com/66761308/182003063-ce81066a-d216-4064-9101-221d9d8dd4eb.PNG)

arpitapandya commented:
> I was not sure how to test this out within my local repository to view it. If anyone could show me how that'd be awesome!

I was doing research on how to preview, I think, by going to the file on your GitHub under website repo from the branch you created, It will allow to preview it, that's how I am seeing if there're no syntax error.

blulady commented:
ETA: 2-3
Availability: 22-8-1

blulady commented:
Hey Justin,
Just wanted to say great work on this issue, there is a lot here. Arpita is right, I am also getting an error when I push your branch up to my github repo and then make this branch the default branch from the branches link under settings .
<details>
<summary> like this </summary>

![Screenshot (206)](https://user-images.githubusercontent.com/67162265/182212217-d5972376-033f-48bb-9e27-4fc8001ae196.jpg)
 
</details>

I am getting this error: 
YAML syntax error: (): did not find expected '-' indicator while parsing a block collection at line 45 column 6. [Learn more](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/common-validation-errors-when-creating-issue-forms).

blulady reviewed (changes requested):
see comment

jdingeman committed:
correct indentation to line 45-47

jdingeman commented:
Hello @blulady and @arpitapandya. I have added the indentation. I couldn't figure out what was wrong with it at first but I had to really hone in one what was wrong with it. 😄

jdingeman committed:
added attributes to lines with checkboxes

jdingeman committed:
Merge branch 'hackforla:gh-pages' into feasibility-issue-form

jdingeman committed:
Update feasibility-issue-template.yml

jdingeman committed:
Merge branch 'feasibility-issue-form' of https://github.com/jdingeman/website into feasibility-issue-form

jdingeman committed:
Update feasibility-issue-template.yml

jdingeman commented:
Hi @arpitapandya and @blulady, I added the indentations and tried to view it through my local repo. I am getting this syntax error

![image](https://user-images.githubusercontent.com/88953806/182479733-a1e68c06-b053-48d4-b1f0-90eb27fce584.png)

Do you know what the expected key should be?

arpitapandya commented:
> Hi @arpitapandya and @blulady, I added the indentations and tried to view it through my local repo. I am getting this syntax error
> 
> ![image](https://user-images.githubusercontent.com/88953806/182479733-a1e68c06-b053-48d4-b1f0-90eb27fce584.png)
> 
> Do you know what the expected key should be?

@jdingeman That error is coming because of the indentation error, can you remove the space before validations on line 51 please. If it gives you error I would suggest remove the line and type it, it's just because of the space you have added by mistake between line 44 to 52
![indentation](https://user-images.githubusercontent.com/66761308/182507102-5898709f-72e0-4ceb-b5db-b5abfda00fe6.PNG).
In case you wanna use the [tool](http://www.yamllint.com/). It will validate.

jdingeman committed:
Update feasibility-issue-template.yml

jdingeman committed:
Update feasibility-issue-template.yml

jdingeman committed:
Update feasibility-issue-template.yml

jdingeman committed:
Update feasibility-issue-template.yml

jdingeman committed:
Update feasibility-issue-template.yml

jdingeman commented:
@arpitapandya thank you so much! I managed to get the file to display as a form on my repo. Let me know what you think. 
PS. Thank you for the tool! It'll be very useful.

blulady commented:
That tool is a delight. Nice find @arpitapandya

blulady reviewed (changes requested):
So it is looking really good!! I think there are only a few more changes that need to be added before this template is complete. Looking at issue #2298 
 - there needs to be a title for the issue that the template will be filled out for
 - there needs to be an input box if you check yes for a dependency

arpitapandya commented:
> So it is looking really good!! I think there are only a few more changes that need to be added before this template is complete. Looking at issue #2298
> 
> * there needs to be a title for the issue that the template will be filled out for
> * there needs to be an input box if you check yes for a dependency

@jdingeman you may add this https://github.com/hackforla/website/pull/3436#pullrequestreview-1054605483 as @blulady suggested to add input box for the dependency label.
- Also, I am not sure, if you were able to use the [extension](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) in VS code, It's such a time saver and catches the error and will guide you what should be removed and allowed.

jdingeman committed:
Update feasibility-issue-template.yml

jdingeman committed:
Update feasibility-issue-template.yml

jdingeman committed:
Update feasibility-issue-template.yml

Removed invalid ids and added validation requirement to dependency dropdown

jdingeman commented:
Hello @blulady and @arpitapandya, I've added the drop down, but I'm not sure if the formatting is correct. Is the drop down used to generate the checklist? Or should the checklist stay there regardless of if "Yes" is selected for "Is there a dependency?"

Also @arpitapandya, thank you for linking that extension! It helped me see some of the unnecessary properties I had added.

blulady commented:
> Hello @blulady and @arpitapandya, I've added the drop down, but I'm not sure if the formatting is correct. Is the drop down used to generate the checklist? Or should the checklist stay there regardless of if "Yes" is selected for "Is there a dependency?"
> 
> Also @arpitapandya, thank you for linking that extension! It helped me see some of the unnecessary properties I had added.
Thanks for working on it! I am on call for PR's on Thursday. I hope someone reviews this before then but I will be reviewing it on Thursday if not.

arpitapandya commented:
> > Hello @blulady and @arpitapandya, I've added the drop down, but I'm not sure if the formatting is correct. Is the drop down used to generate the checklist? Or should the checklist stay there regardless of if "Yes" is selected for "Is there a dependency?"
> > Also @arpitapandya, thank you for linking that extension! It helped me see some of the unnecessary properties I had added.
> > Thanks for working on it! I am on call for PR's on Thursday. I hope someone reviews this before then but I will be reviewing it on Thursday if not.

- For the dependency label my understanding is if you can make it like [example form](https://github.com/hackforla/website/blob/gh-pages/.github/ISSUE_TEMPLATE/blank-issue-form.yml) as stated in issue `then an input for what it is if yes)`
- Can you move dependency after title (first title, Dependency and then Overview in order as per the issue)?
- For the Action Items and Checklist I saw you made changes, I would love to learn from @blulady before I jump and make an assumption as I am not too sure.
Again, Thanks @jdingeman for constantly following up and working on this issue by making changes.

jdingeman committed:
Change dependency section

jdingeman committed:
demo textarea for action items and feasibility

jdingeman commented:
@blulady and @arpitapandya, now that I look at it, I am curious if the action items and instructions should be checklists by default, or appear as textarea for user input like the [blank form](https://github.com/hackforla/website/blob/gh-pages/.github/ISSUE_TEMPLATE/blank-issue-form.yml)

<details>
<summary>With checklists</summary>
<img src="https://user-images.githubusercontent.com/88953806/184028259-c2fe70db-7ab0-4e0b-ab67-51f8bcddfa4a.png">
</details>

<details>
<summary>With textareas</summary>
<img src="https://user-images.githubusercontent.com/88953806/184029557-862548d7-c355-45e7-b738-0d49f388a163.png">
</details>

#2298 uses [this form](https://github.com/hackforla/website/issues/2128#issue-973987353) as an example, so I imagine they both would need checklists instead of textareas.

jdingeman committed:
Merge branch 'hackforla:gh-pages' into feasibility-issue-form

arpitapandya commented:
> @blulady and @arpitapandya, now that I look at it, I am curious if the action items and instructions should be checklists by default, or appear as textarea for user input like the [blank form](https://github.com/hackforla/website/blob/gh-pages/.github/ISSUE_TEMPLATE/blank-issue-form.yml)
> 
> With checklists
> ![](https://user-images.githubusercontent.com/88953806/184028259-c2fe70db-7ab0-4e0b-ab67-51f8bcddfa4a.png)
> With textareas
> ![](https://user-images.githubusercontent.com/88953806/184029557-862548d7-c355-45e7-b738-0d49f388a163.png)
> #2298 uses [this form](https://github.com/hackforla/website/issues/2128#issue-973987353) as an example, so I imagine they both would need checklists instead of textareas.

@jdingeman So, the [issue](https://github.com/hackforla/website/issues/2298) indicates that the resulting form should be as [form](https://github.com/hackforla/website/issues/2128), consider it as `.md` and creating `.yml` version for that.
- I don't think `textarea` will generate checklist for how the feasibility tests should look like. I would make as listed in the demo.
- correct me if I am wrong @blulady

blulady commented:
> @blulady and @arpitapandya, now that I look at it, I am curious if the action items and instructions should be checklists by default, or appear as textarea for user input like the [blank form](https://github.com/hackforla/website/blob/gh-pages/.github/ISSUE_TEMPLATE/blank-issue-form.yml)
> 
> With checklists
> ![](https://user-images.githubusercontent.com/88953806/184028259-c2fe70db-7ab0-4e0b-ab67-51f8bcddfa4a.png)
> With textareas
> ![](https://user-images.githubusercontent.com/88953806/184029557-862548d7-c355-45e7-b738-0d49f388a163.png)
> #2298 uses [this form](https://github.com/hackforla/website/issues/2128#issue-973987353) as an example, so I imagine they both would need checklists instead of textareas.

That is a lot of work @jdingeman. You have done a great job getting to know how to create an issue template in a few different ways. Your template has a title, a dependency drop down (which looks like the [form](https://github.com/akibrhast/website/issues/new?assignees=octocat&labels=bug%2Ctriage&template=bug.yml&title=%5BBug%5D%3A+) cited in the issue your issue cites and while it is not technically a "checkbox" it does match the example), there is an overview text area, there is a resources/instructions text area and while there is a feasibility area it is not a checklist. Personally I like checkboxes because they can check off what is done but these won't be as specific and therefore not as helpful
I feel like there is room for interpretation because instead of linking a specific form for an example, an entire other issue is linked. 
I think it would be fine approving this issue but we could also bring it up to Bonnie in Sunday's meeting to get clarification. You've put in a ton of work on this so I will leave it up to you.

blulady commented:
> > @blulady and @arpitapandya, now that I look at it, I am curious if the action items and instructions should be checklists by default, or appear as textarea for user input like the [blank form](https://github.com/hackforla/website/blob/gh-pages/.github/ISSUE_TEMPLATE/blank-issue-form.yml)
> > With checklists
> > ![](https://user-images.githubusercontent.com/88953806/184028259-c2fe70db-7ab0-4e0b-ab67-51f8bcddfa4a.png)
> > With textareas
> > ![](https://user-images.githubusercontent.com/88953806/184029557-862548d7-c355-45e7-b738-0d49f388a163.png)
> > #2298 uses [this form](https://github.com/hackforla/website/issues/2128#issue-973987353) as an example, so I imagine they both would need checklists instead of textareas.
> 
> That is a lot of work @jdingeman. You have done a great job getting to know how to create an issue template in a few different ways. Your template has a title, a dependency drop down (which looks like the [form](https://github.com/akibrhast/website/issues/new?assignees=octocat&labels=bug%2Ctriage&template=bug.yml&title=%5BBug%5D%3A+) cited in the issue your issue cites and while it is not technically a "checkbox" it does match the example), there is an overview text area, there is a resources/instructions text area and while there is a feasibility area it is not a checklist. I think that the feasibility check boxes will need a text area below them for specification. There is room for interpretation because instead of linking a specific form for an example, an entire other issue is linked. I think it would be fine approving this issue once we add the feasibility checkboxes but we could also bring it up to Bonnie in Sunday's meeting to get clarification. You've put in a ton of work on this so I will leave it up to you.

blulady commented:
I do think keeping the text area for

blulady commented:
I think that the feasibility check boxes will need a text area below them for specification.

blulady reviewed (changes requested):
see comment.

arpitapandya commented:
> > > Hello @blulady and @arpitapandya, I've added the drop down, but I'm not sure if the formatting is correct. Is the drop down used to generate the checklist? Or should the checklist stay there regardless of if "Yes" is selected for "Is there a dependency?"
> > > Also @arpitapandya, thank you for linking that extension! It helped me see some of the unnecessary properties I had added.
> > > Thanks for working on it! I am on call for PR's on Thursday. I hope someone reviews this before then but I will be reviewing it on Thursday if not.
> 
> * For the dependency label my understanding is if you can make it like [example form](https://github.com/hackforla/website/blob/gh-pages/.github/ISSUE_TEMPLATE/blank-issue-form.yml) as stated in issue `then an input for what it is if yes)`
> * Can you move dependency after title (first title, Dependency and then Overview in order as per the issue)?
> * For the Action Items and Checklist I saw you made changes, I would love to learn from @blulady before I jump and make an assumption as I am not too sure.
>   Again, Thanks @jdingeman for constantly following up and working on this issue by making changes.

@jdingeman I am so sorry for the misunderstanding that I misspelled and linked `example form` that may caused you to work more and do all changes. 
- when I referenced `example form` was just for the dependency label as that template form had that label. I didn't mean to make you do all changes and make that same form again, as I know you already did make correct form, just that Dependency label was missing partially. 
-  I referenced as the past issue has lots of sample forms and demos created by other developers in past, which we can use as a resource for certain pieces, but I was aware that there's already `resulting issue form` linked to that which we're focusing on.
- As you have worked a lot, I would leave it up to you and @blulady

jdingeman committed:
insert checklist above text area for feasibility

jdingeman committed:
Update feasibility-issue-template.yml

jdingeman commented:
@arpitapandya no worries at all! 
@blulady do you mean like this? 
![image](https://user-images.githubusercontent.com/88953806/184566997-4653a7f5-2525-4653-8f74-545f11016ab0.png)

jdingeman committed:
Changed action items to markdown format

jdingeman committed:
remove quotations + comment out

jdingeman commented:
Leaving the commented sections in to easily switch between checklist/textarea format depending on what is best suited for the issue form.

blulady commented:
> @arpitapandya no worries at all! @blulady do you mean like this? ![image](https://user-images.githubusercontent.com/88953806/184566997-4653a7f5-2525-4653-8f74-545f11016ab0.png)

That looks good to me. I might go with Feasibility item 1, 2, 3.

blulady reviewed (commented):
So you have totally fulfilled the requirements of the task but I was thinking that at the bottom of the checklists there would be an area for someone to list what the items are. So there would be a checklist for action items at the bottom of the checklist to say what the checklist is. Same with feasibility items. However, you have technically fulfilled the requests in the linked issue so I'll leave it up to you. Just let me know what you want to do with it
![Screenshot (216)](https://user-images.githubusercontent.com/67162265/185455787-6067cfb2-ec0f-43c3-ac20-6b3c916b1e19.jpg)
.

jdingeman commented:
> 

I will go ahead and add the textareas. I agree with you on this.

jdingeman committed:
Remove comments, add text areas

jdingeman committed:
Change textarea labels

jdingeman committed:
Change textarea labels

blulady reviewed (approved):
Looks Good!!! The PR is coming from the correct branch and going into the correct branch. It cites the correct issue and addresses that issue. There are only changes in the feasibility-issue-template.yml, even though there are 27 commits. I am stoked to merge this issue. Good job @jdingeman
